### PR TITLE
Java 11 compatibility

### PR DESCRIPTION
**Pull request by timja**
👋 Noticed these tests didn't pass on my machine.

Mostly because of heavy use of powermock which requires some updates to work on Java 11.

I think powermock could be done away with fairly easily and in general is the preferred approach normally, but I've done the smallest fix to get this working on java 11

### CODE REVIEW

File: Jenkinsfile
Comment (timja) [inline Jenkinsfile:2]: this change won't actually be run on this PR as I don't have write access to the repo, but I've ran the build locally and it passes now

File: pom.xml
Comment (timja) [inline pom.xml:67]: the current baseline was too old for java 11 support, I selected a more modern one that users will be using based on https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/#currently-recommended-versions

File: src/test/java/jenkins/plugins/logstash/LogstashConfigurationMigrationTest.java
Comment (timja) [inline src/test/java/jenkins/plugins/logstash/LogstashConfigurationMigrationTest.java:38]: https://wiki.jenkins.io/display/JENKINS/Java+11+developer+guidelines refers to this as the standard powermock fix

File: src/test/java/jenkins/plugins/logstash/LogstashConfigurationMigrationTest.java
Comment (timja) [inline src/test/java/jenkins/plugins/logstash/LogstashConfigurationMigrationTest.java:55]: some weird powermock issue (https://github.com/powermock/powermock/issues/992), this fix worked

File: src/test/java/jenkins/plugins/logstash/persistence/ElasticSearchSSLCertsTest.java
Comment (timja) [inline src/test/java/jenkins/plugins/logstash/persistence/ElasticSearchSSLCertsTest.java:120]: error message changes quite a lot on java 11...

File: src/test/java/jenkins/plugins/logstash/LogstashIntegrationTest.java
Comment (timja) [inline src/test/java/jenkins/plugins/logstash/LogstashIntegrationTest.java:76]: jenkins log now has 5 messages at this point, I assume you don't really care how many messages there are

File: Jenkinsfile
Comment (jakub-bochenski) [reply Jenkinsfile:2]: Given that Java 8 is EOL it might make sense to drop it from testing

**Comment by jakub-bochenski** [comment]
LGTM I just need to think if Java 8 should be dropped

File: Jenkinsfile
Comment (timja) [reply Jenkinsfile:2]: it's not really, each vendor can choose how long they will support it for, oracle has chosen to stop doing updates for commercial use although personal and developer updates are still there for free,

adoptopenjdk is saying 2026 for their distribution: https://adoptopenjdk.net/support.html

When Jenkins drops java 8 as a requirement (hopefully sometime next year) then it would make sense to do this but not yet

https://groups.google.com/g/jenkinsci-dev/c/nEw4TLDcZLw/m/x5KJ0WVFAAAJ

**Comment by timja** [comment]
> LGTM I just need to think if Java 8 should be dropped

https://github.com/jenkinsci/logstash-plugin/pull/97#discussion_r549674801

File: Jenkinsfile
Comment (jakub-bochenski) [reply Jenkinsfile:2]: Makes sense, thanks for the reference

**Review by jakub-bochenski — APPROVED** [review]

**Pull request merged**
